### PR TITLE
Move neoboard to own i18n namespace

### PR DESCRIPTION
**Pull request by MTRNord**
Required for NEO-711

**:heavy_check_mark: Checklist**

- [ ] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [ ] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,5 +68,6 @@
     "*.{yaml,yml,json,md}": [
       "prettier --write"
     ]
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha1.ac34549e6aa8e7ead463a7407e1c7390f61a6610"
```
[the remark this answers is not in the document]
Comment (MTRNord) [reply package.json:72]: sideeffect of me using https://github.com/nodejs/corepack due to draupnir. Gets added as soon as I invoke yarn ^^ Similar to the ecosystem thingy at the top of the file essentially. Shouldnt harm anyone but basically tells corepack users which yarn version has to be used.

File: packages/react-sdk/src/App.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ export type AppProps = {
 };
 
 export const App = ({ layoutProps }: AppProps) => {
-  const { t } = useTranslation();
+  const { t } = useTranslation('neoboard');
```
[the remark this answers is not in the document]
Comment (MTRNord) [reply packages/react-sdk/src/App.tsx:28]: So the thing is that the default namespace is for the current i18n namespace. It works for the widget itself of course but for any application using the react-sdk they will likely have their own default namespace which is different. Meaning we sadly have to do this for it to be consumable. The default namespace is essentially shared across all dependencies consuming the i18n project. Its not scoped per project.

[the remark this answers is not in the document]
Comment (HarHarLinks) [reply packages/react-sdk/src/App.tsx:28]: I can see the default being set in the widget's init, am not sure what it's needed in the react-sdk exports for, but I get why it would need to be hardcoded somewhere in the react-sdk.

[the remark this answers is not in the document]
Comment (MTRNord) [reply packages/react-sdk/src/App.tsx:28]: > I can see the default being set in the widget's init, am not sure what it's needed in the react-sdk exports for, but I get why it would need to be hardcoded somewhere in the react-sdk.

You can either set it on the i18n init which sets it for the whole i18n instance. Thats whats done on the widget.

Then you can set it on each hook which is local for the hook. This is required as we cant assume the default namespace of consumers. The widget is one consumer but it isn't the only consumer. So we have to make sure that it is still using the neoboard namespace for neoboard translations regardless of the default namespace. We can (arguably) remove the default namespace on the widget itself. Its just there since it would kinda make sense. It wont have any significant effect however.

TLDR:

We cant assume things about how the i18n instance of consumers is set up. Hence we need to use i18next in predictable ways instead.

[the remark this answers is not in the document]
Comment (MTRNord) [reply packages/react-sdk/src/App.tsx:28]: grafik

Graphic that hopefully makes this a bit clearer.

So both things always access one i18n instance. However the namespaces may be different
